edx_proctoring/signals.py

```diff
@@ -105,6 +106,18 @@ def on_attempt_changed(sender, instance, signal, **kwargs):  # pylint: disable=u
             # to archive it
             original = sender.objects.get(id=instance.id)
 
+            # if the exam was finished for the first time, we want to mark it as
+            # complete in the Completion Service.
+            # This functionality was added because regardless of submission status
+            # on individual problems, we want to mark the entire exam as complete
+            # when the exam is finished since there are no more actions a learner can take.
+            if not original.completed_at and instance.completed_at:
```

Comment:
Is it difficult to have unit test covering the code change?

Reply:
So I'm honestly not super sure. I didn't see any tests for the existing signals that are calling out so I wasn't sure how this repo wanted to handle that. I'm happy to try and figure it out though.

Reply:
I would feel more comfortable with a unit test. Also, it would improve the codecov check too.

Reply:
Okay, happy to figure that out